go-controller/pkg/ovn/controller/apbroute/external_controller.go

```diff
@@ -565,10 +565,13 @@ func (m *externalPolicyManager) onPodUpdate(oldObj, newObj interface{}) {
 		utilruntime.HandleError(errors.New("invalid Pod provided to onPodUpdate()"))
 		return
 	}
-	// if labels AND assigned Pod IPs AND the multus network status annotations are the same, skip processing changes to the pod.
+	// if labels AND assigned Pod IPs AND the multus network status annotations AND
+	// pod status conditions AND deletion timestamp are the same, skip processing changes to the pod.
 	if reflect.DeepEqual(o.Labels, n.Labels) &&
 		reflect.DeepEqual(o.Status.PodIPs, n.Status.PodIPs) &&
-		reflect.DeepEqual(o.Annotations[nettypes.NetworkStatusAnnot], n.Annotations[nettypes.NetworkStatusAnnot]) {
+		reflect.DeepEqual(o.Annotations[nettypes.NetworkStatusAnnot], n.Annotations[nettypes.NetworkStatusAnnot]) &&
+		reflect.DeepEqual(o.Status.Conditions, n.Status.Conditions) &&
```

Comment:
pod's conditions could fluctuate right? is this what is used to tell if the pod went unhealthy?
why did we add deletion timestamp here ? can you add a comment behind why you added the new match conditions?

Reply:
I have added the reason for the new matches. PLMK if I should add more details.

Reply:
I have one last question here.. shouldn't we be explicitly then checking if the podready status changed? checking the whole of status conditions means we could get a lot more noise than we actually care about? so i'd rather you check on pod ready v/s not ready

Reply:
so it should be podBecameReady || podBecameNotReady

Reply:
Ack. Will update the code to only reconcile the pod when the ready status changes.

Reply:
Updated the code to check the changes in PodReady status condition.